### PR TITLE
add a new Score db obj, keyed on email with a streak col

### PR DESCRIPTION
**Pull request by kzfubar**

### CODE REVIEW

File: shame_script.py [hunk 1]
```diff
@@ -151,8 +161,9 @@ async def fetch_and_send_tasks() -> None:
                 )
 
                 message_content.append(
-                    f"*Tasks for {discord_user.mention}*\n```\n{table}\n```"
+                    f"*Tasks for {discord_user.mention}* | Streak: {user_score.streak}\n```\n{table}\n```"
```
Comment (rocket3989) [inline shame_script.py]: breaking the formatting in the middle of the line doesn't look great, maybe should be in a new line?

File: shame_script.py [hunk 2]
```diff
@@ -151,8 +161,9 @@ async def fetch_and_send_tasks() -> None:
...
                 )
+            database_session.commit()
```
Comment (rocket3989) [inline shame_script.py:163]:
```suggestion
```
`.commit` is run as part of `with` context exit

File: utils/Database.py [hunk 3]
```diff
@@ -28,6 +28,15 @@ def __repr__(self) -> str:
...
 
+class Score(Base):
+    __tablename__ = "scores"
+    email: Mapped[str] = mapped_column(primary_key=True)
+    streak: Mapped[int] = mapped_column()
+
+    def __repr__(self) -> str:
+        return f"<Score(email={self.email}, streak={self.streak})>"
+
+
```
Comment (rocket3989) [inline utils/Database.py]: I don't hate this value just being added to the primary table, but if it is it's own table, it should follow ORM mapping: https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#one-to-one

**Review by rocket3989 — CHANGES_REQUESTED** [review]
added comments

File: shame_script.py (hunk 1 above)
Comment (kzfubar) [reply shame_script.py]: i think adding a newline for streak looks messier

heyitsme completed all tasks!
Streak: 1
rocket completed all tasks!
Streak: 1
snackwel completed all tasks!
Streak: 1

is doubling the size of this message, and there association between streak and user is a little less clear

maybe we can just keep the formatting?

File: shame_script.py (hunk 2 above)
Comment (kzfubar) [reply shame_script.py:163]: do you have a source for this?

empirically testing this is not the case

File: utils/Database.py (hunk 3 above)
Comment (kzfubar) [reply utils/Database.py]: i'm still leaning for a seperate table, i think a category of points / score related things can exist outside of user logisitcs?

added one to one ORM mapping

File: shame_script.py (hunk 2 above)
Comment (rocket3989) [reply shame_script.py:163]: I was wrong

File: shame_script.py (hunk 1 above)
Comment (rocket3989) [reply shame_script.py]: keeping the formatting seems fine

**Comment by rocket3989** [comment]
switch to using ORM accessors (`user.score.x`) . Fixing the existing User table is optional, can keep the relation as is

**Review by rocket3989 — APPROVED** [review]

**Comment by rocket3989** [comment]
@kzfubar please take a look at the changes I made

File: shame_script.py [hunk 4]
```diff
@@ -56,7 +56,6 @@ async def on_ready() -> None:
     try:
         synced = await bot.tree.sync()
         await fetch_and_send_tasks()
-        fetch_and_send_tasks.start()
```
Comment (kzfubar) [inline shame_script.py:58]: whoops

**Pull request merged**